clang/lib/Driver/ToolChain.cpp

```diff
@@ -922,6 +923,22 @@ bool ToolChain::isThreadModelSupported(const StringRef Model) const {
   return false;
 }
 
+void ToolChain::setDecimalFloatABI(const llvm::opt::ArgList &Args,
+                                   const llvm::Triple &Triple) const {
+  tools::DecimalFloatABI ABI =
+      tools::getDecimalFloatABI(getDriver(), Triple, Args);
+  tools::DecimalFloatABI DefaultDecimalABI =
+      tools::getDefaultDecimalFloatABI(Triple);
+  if (DefaultDecimalABI != tools::DecimalFloatABI::None &&
+      ABI != DefaultDecimalABI) {
+    Arg *ABIArg =
+        Args.getLastArg(options::OPT_mdecimal_float_abi_EQ);
+    assert(ABIArg && "Non-default decimal float abi expected to be from arg");
+    D.Diag(diag::err_drv_unsupported_opt_for_target)
+        << ABIArg->getAsString(Args) << Triple.getTriple();
+  }
```

Comment:
Not really sure I understand your suggestion here. Are you proposing that a virtual function be added to `TargetInfo`? From here I don't have any way of getting to the `TargetInfo`. The decimal floating-point ABI needs to be computed here so that I can fill up the `cc1` option. Unless I am not seeing things clearly?